mm2src/coins/lp_coins.rs

```diff
@@ -2707,23 +2739,103 @@ impl Default for PrivKeyActivationPolicy {
     fn default() -> Self { PrivKeyActivationPolicy::ContextPrivKey }
 }
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub enum PrivKeyPolicy<T> {
-    KeyPair(T),
+    Iguana(T),
+    HDWallet {
+        /// Derivation path of the coin.
+        /// This derivation path consists of `purpose` and `coin_type` only
+        /// where the full `BIP44` address has the following structure:
+        /// `m/purpose'/coin_type'`.
+        derivation_path: StandardHDPathToCoin,
+        activated_key: T,
+        bip39_secp_priv_key: ExtendedPrivateKey<secp256k1::SecretKey>,
+    },
     Trezor,
+    #[cfg(target_arch = "wasm32")]
+    Metamask(EthMetamaskPolicy),
+}
+
+#[cfg(target_arch = "wasm32")]
+#[derive(Clone, Debug)]
+pub struct EthMetamaskPolicy {
```

Comment:
It's a little doubtful to move eth related concept to the common module. May be it could be played out with something like that to hide less common concepts:

```
    #[cfg(target_arch = "wasm32")]
    Metamask(T),
```